features/FEATURE_LWIP/lwip-interface/ppp_lwip.cpp

```diff
@@ -18,6 +18,14 @@
 #include "platform/FileHandle.h"
 #include "platform/mbed_poll.h"
 #include "events/EventQueue.h"
+#if defined(FEATURE_COMMON_PAL)
+#include "mbed_trace.h"
+#define TRACE_GROUP "LPPP"
+#else
+#define tr_debug(...) (void(0)) //dummies if feature common pal is not added
```

Comment:
ATParser have traditionally been using debug_if. We changed code only if it was totally necessary, otherwise we kept our paws off off the code.    We introduced mbed-trace in PPP's implementation as LWIP's debug traces were consuming way too much RAM and were not well aligned.

What do you mean by "Conditional FEATURE_COMMON_PAL is not acceptable" ?
Its just a macro test, which tells me if I can use mbed-trace or not.